dbt/adapters/athena/impl.py

```diff
@@ -1262,6 +1266,47 @@ def format_partition_keys(self, partition_keys: List[str]) -> str:
 
     @available
     def format_one_partition_key(self, partition_key: str) -> str:
-        """Check if partition key uses Iceberg hidden partitioning"""
+        """Check if partition key uses Iceberg hidden partitioning or bucket partitioning"""
         hidden = re.search(r"^(hour|day|month|year)\((.+)\)", partition_key.lower())
-        return f"date_trunc('{hidden.group(1)}', {hidden.group(2)})" if hidden else partition_key.lower()
+        bucket = re.search(r"bucket\((.+),", partition_key.lower())
+        if hidden:
+            return f"date_trunc('{hidden.group(1)}', {hidden.group(2)})"
+        elif bucket:
+            return bucket.group(1)
+        else:
+            return partition_key.lower()
+
+    @available
+    def murmur3_hash(self, value: Any, num_buckets: int) -> Any:
+        """Computes a hash for the given value using the MurmurHash3 algorithm and returns a bucket number."""
```

Comment:
It would be nice to add link to the source where this method was adopted from. Is it this [method](https://github.com/apache/iceberg-python/blob/088ee4076e733422d7d50491cab8860ae08426af/pyiceberg/transforms.py#L240C12-L240C12)  ?

Reply:
Done ✅